keps/sig-scheduling/4120-pod-topology-spread-persisten-volume-claim/README.md

```diff
@@ -0,0 +1,851 @@
+
+The root cause is that PodTopologySpread doesn't include deleted pod filter and score process.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- Allow deleted pod with node affinity `PersistentVolumeClaim` participate in `PodTopologySpread` plugin 
+  filter and score process.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+When calculating pod topology spread skew, I want to exclude nodes that have node affinity volume of terminated pod to prevent recreate termianted pods from falling into unexpected Pending state.
+
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+- Checking pod whether exist depend on PersistentVolumeClaim may lead to performance problem, we need to verify this by adding performance tests. If performance problem does exists, we'd like to add the process of determining whether a pod exists during PreFilter.
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+This KEP proposes a method that use `volume.kubernetes.io/selected-node` annotation and ownerReferences
+of `PersistentVolumeClaim` to replace deleted pod participate in `PodTopologySpread` plugin filter and score process.
```

Comment:
Can you specify exactly what the scheduler algorithm changes will be?

Reply:
Take the `PodTopologySpread` scheduling plugin as an example. In the `Prefilter` and `PreScore` stages, the deleted pods need to be infered according to the ownerReferences of pvc, so that these pods can participate in the construction of `preFilterState` and `preScoreState`. more detail please look [PR](https://github.com/kubernetes/kubernetes/pull/119119) associated with this proposal